xdsl/dialects/snitch_runtime.py

```diff
@@ -52,6 +52,125 @@ def __init__(
         super().__init__(operands=[], result_types=[])
 
 
+@irdl_op_definition
+class GlobalCoreBaseHartidOp(SnitchRuntimeGetInfo):
+    """ """
+
+    name = "snrt.global_core_base_hartid"
+
+
+@irdl_op_definition
+class GlobalCoreIdxOp(SnitchRuntimeGetInfo):
+    """
+    Core ID of each core on all clusters
+    """
+
+    name = "snrt.global_core_idx"
+
+
+@irdl_op_definition
+class GlobalCoreNumOp(SnitchRuntimeGetInfo):
+    """
+    Total cores incl. DM core per cluster
+    """
+
+    name = "snrt.global_core_num"
+
+
+@irdl_op_definition
+class GlobalComputeCoreIdxOp(SnitchRuntimeGetInfo):
+    """ """
```

Comment:
Please add some docs, these are quite important to understand the overall cluster topology.